org.jacoco.core.test/src/org/jacoco/core/test/InstrumentingLoader.java

```diff
@@ -21,31 +20,53 @@
 import org.jacoco.core.runtime.RuntimeData;
 import org.jacoco.core.runtime.SystemPropertiesRuntime;
 
+/**
+ * Class loader which loads classes from another class loader and instruments
+ * them.
+ */
 public final class InstrumentingLoader extends ClassLoader {
 
-	private final RuntimeData data;
 	private final IRuntime runtime;
+	private final String scope;
+	private final ClassLoader delegate;
 
-	public InstrumentingLoader() throws Exception {
-		data = new RuntimeData();
-		runtime = new SystemPropertiesRuntime();
+	private final RuntimeData data;
+	private final Instrumenter instrumenter;
+
+	public InstrumentingLoader(IRuntime runtime, String scope,
+			ClassLoader delegate) throws Exception {
+		this.runtime = runtime;
+		this.scope = scope;
+		this.delegate = delegate;
+		this.data = new RuntimeData();
 		runtime.startup(data);
+		this.instrumenter = new Instrumenter(runtime);
+	}
+
+	public InstrumentingLoader(Class<?> target) throws Exception {
+		this(new SystemPropertiesRuntime(), target.getPackage().getName(),
+				target.getClassLoader());
+	}
+
+	public InstrumentingLoader() throws Exception {
```

Comment:
Right, I removed the too specific default constructor. Still want to keep the constructor where all delegates can be injected to enable this utility for future scenarios (other runtimes, other class loaders).